pyarbtools/instruments.py

```diff
@@ -1401,36 +1541,46 @@ def bin_pdw_file_builder(self, pdwList):
             (bytes): Binary data that contains a full PDW file that can
                 be downloaded to and played out of the UXG.
         """
+        #Include frequency phase coding block flag: 1 = yes, 0 = no
+        includeFpcBlock = 1
 
         # Header section, all fixed values
         fileId = b'STRM'
         version = (1).to_bytes(4, byteorder='little')
-        # No reason to have > one 4096 byte offset to PDW data.
-        offset = ((1 << 1) & 0x3fffff).to_bytes(4, byteorder='little')
+
+        # First field is first block of 4096 bytes.  If frequency phase coding block is large,
+        # this offset to the start of PDW data might extend past first 4096 sized block
+        fieldBlock = 1
+        offset = ((fieldBlock >> 1) & 0x3fffff).to_bytes(4, byteorder='little')
+
         magic = b'KEYS'
         res0 = (0).to_bytes(16, byteorder='little')
         flags = (0).to_bytes(4, byteorder='little')
         uniqueId = (0).to_bytes(4, byteorder='little')
         dataId = (16).to_bytes(4, byteorder='little')
         res1 = (0).to_bytes(4, byteorder='little')
         header = [fileId, version, offset, magic, res0, flags, uniqueId, dataId, res1]
+        tempHeaderSize = len(b''.join(header))
 
-        # Padding block, all fixed values
-        padBlockId = (1).to_bytes(4, byteorder='little')
-        res3 = (0).to_bytes(4, byteorder='little')
-        size = (4016).to_bytes(8, byteorder='little')
-        # 4016 bytes of padding ensures that the first PDw begins @ byte 4097
-        padData = (0).to_bytes(4016, byteorder='little')
-        padding = [padBlockId, res3, size, padData]
+        # FPC Block - skip fpcBlock if flag is zero
+        fpcBlock = [b'']
+        if (includeFpcBlock):
+            fpcBlock = self.bin_pdw_freqPhaseCodingBlock()
+        fpcBlockSize = len(b''.join(fpcBlock))
 
-        # PDW block
+        #PDW block header must start at byte 4080 so PDW stream data starts at byte 4097
+        paddingSize = 4080 - tempHeaderSize - fpcBlockSize
+        paddingBlock = self.paddingBlock(paddingSize)
+
+        # PDW block header = 16 bytes
         pdwBlockId = (16).to_bytes(4, byteorder='little')
         res4 = (0).to_bytes(4, byteorder='little')
         pdwSize = (0xffffffffffffffff).to_bytes(8, byteorder='little')
         pdwBlock = [pdwBlockId, res4, pdwSize]
 
         # Build PDW file from header, padBlock, pdwBlock, and PDWs
-        pdwFile = header + padding + pdwBlock
+        pdwFile = header + fpcBlock + paddingBlock + pdwBlock
+
```

Comment:
This is an elegant solution to adding the FPC block into a PDW. Nice work.

Reply:
Thanks. We could also incorporate the padding block method into bin_pdw_file_builder in the VectorUXG class in instruments.py as well.  I didn't want to make changes without consulting you first.

Reply:
I think that's a good idea, but let's add it in once we've fully addressed the structure of defining the FPC and incorporating it into generic PDWs.